### PR TITLE
Gutenberg: Apply box-sizing: border-box to the entire section

### LINKED ISSUE #27651: Gutenberg: CSS box-sizing
A follow up to #27525

Let's see if we can get some cleaner CSS fixes by defaulting this section to use box-sizing border-box. As noted by @copons: 

> Just wanted to say that, apparently, lots of the glitches we're having with Gutenberg in Calypso are caused by the different box-sizing. Gutenberg defaults everything to border-box, while Calypso doesn't make such assumption and leaves the freedom to define it as needed.

### PR DESCRIPTION
**Pull request by Copons**
#### Changes proposed in this Pull Request

Alt approach to #27914

* Apply `box-sizing: border-box` to the entire Gutenberg section

#### Testing instructions

* Smoke test Gutenberg at `http://calypso.localhost:3000/gutenberg/post/`, and see that most things look good (probably right now only the checkboxes and radio buttons in the sidebar looks wrong).

Fixes #27651

### CODE REVIEW

File: client/gutenberg/editor/style.scss [hunk 1]
```diff
@@ -108,28 +100,7 @@ $gutenberg-theme-toggle: #11a0d2;
...
 			top: 5px;
 			width: auto;
-			z-index: 100000; /* Above WP toolbar. */
-		}
-	}
-
-	.editor-block-list__block {
-		ul ul, ol ol {
-			list-style-type: circle;
+			z-index: 100000; // Above WP toolbar.
```
Comment (vindl) [inline client/gutenberg/editor/style.scss]: its-over-9000-gif-1

File: client/gutenberg/editor/style.scss [hunk 2]
```diff
@@ -108,28 +100,7 @@ $gutenberg-theme-toggle: #11a0d2;
...
 		}
 	}
 }
-
-//needed for oembed iframes to appear
-.wp-block-embed__wrapper > iframe {
```
Comment (vindl) [inline client/gutenberg/editor/style.scss]: Do we still need this, because `box-sizing` updates don't seem related?

Comment (Copons) [reply client/gutenberg/editor/style.scss]: I actually tested this by following the instructions included in #27486 and I couldn't see any problems with the Oembed block.
@gwwar do you happen to remember if there are corner cases to check here?

Comment (vindl) [reply client/gutenberg/editor/style.scss]: I went through the original testing steps too and confirmed that it works as expected even without these styles. 👍Might be the case that package updates made this fix obsolete.
